src/query.rs

```diff
@@ -470,6 +483,40 @@ impl WMIConnection {
 
         self.raw_query(&query)
     }
+
+    // Execute the given query in async way, returns result in a Sink.
+    #[cfg(feature = "async-query")]
+    fn exec_async_query_native_wrapper(
+        &self,
+        query: impl AsRef<str>,
+    ) -> Result<ComPtr<IWbemObjectSink>, WMIError> {
+        let query_language = BStr::from_str("WQL")?;
+        let query = BStr::from_str(query.as_ref())?;
+
+        unsafe {
+            // ExecQueryAsync needs a *mut ptr, so we encapsulate in a Box to keep a ref to the sink
+            let pp_sink = Box::new(QuerySink::new_ptr());
```

Comment:
I'm just assuming, but if you return a `ComPtr` from `new`, you could use
https://github.com/retep998/wio-rs/blob/master/src/com.rs#L93
which seems more correct overall?